### PR TITLE
Made test_vcache_flush test lowest and highest word

### PR DESCRIPTION
**Pull request by save-buffer**
Addresses part of #451

### CODE REVIEW

**Comment by save-buffer** [comment]
Not long. Timing it gives this:
`make test_vcache_flush.log  3.96s user 0.16s system 100% cpu 4.099 total`

**Comment by drichmond** [comment]
Can you raise the random address limit, and make it reproducible using a seed?

**Comment by save-buffer** [comment]
Limit is now 128, seed is DEADBEEF. It takes ~64 seconds to run now in cosim.

**Comment by tommydcjung** [comment]
can we also have tests to byte/half size access?

**Comment by drichmond** [comment]
Thanks @tommydcjung, this is a good suggestion

**Comment by tommydcjung** [comment]
If we could include tests to byte/half access in tile DMEM, that would also be helpful.

**Comment by drichmond** [comment]
Also a good idea, but that would be the subject of another PR (one feature per PR policy)

**Comment by drichmond** [comment]
@tommydcjung Can you file an issue for that? We'll do it

**Comment by save-buffer** [comment]
Ok so I should do byte/half word testing from vcache in this PR and then from DMEM in another PR?

**Comment by drichmond** [comment]
yep! Separate PR and separate test

**Comment by drichmond** [comment]
@mrutt92 When you have a moment, can you add some feedback as our resident C++ guru?

File: regression/library/test_vcache_flush.cpp [hunk 1]
```diff
@@ -169,15 +190,26 @@ int test_vcache_flush() {
 
         addr_bitwidth = hb_mc_config_get_network_bitwidth_addr(config);
         addrs[0] = 0;
-        addrs[1] = 1 << (addr_bitwidth-2); // Convert to word-level addr
+        addrs[NUM_TESTS - 1] = (1 << (addr_bitwidth-1)) - 1; // Word-level addr of highest word
```
Comment (mrutt92) [inline regression/library/test_vcache_flush.cpp]: Should this be `(1 << (addr_bitwdith)) - 1` ?

File: libraries/bsg_manycore.cpp [hunk 2]
```diff
@@ -1277,9 +1277,11 @@ static int hb_mc_manycore_send_read_rqst(hb_mc_manycore_t *mc,
                 hb_mc_request_packet_set_mask(&rqst.request, HB_MC_PACKET_REQUEST_MASK_WORD);
```
Comment (mrutt92) [inline libraries/bsg_manycore.cpp]: @save-buffer I think you're already aware of this but the masks need to be set according to which bytes are being read. The NPA contains a byte address that should be used for byte selection.

File: regression/library/test_vcache_flush.cpp [hunk 3]
```diff
@@ -169,15 +190,26 @@ int test_vcache_flush() {
...
+        uint32_t data32 = rand();
+        uint16_t data16 = rand() & 0xFFFF;
+        uint8_t   data8 = rand() & 0xFF;
 
-        for (dram = 0; dram < ndrams; ++dram){
+        for (dram = 0; dram < ndrams; ++dram) {
                 dram_coord_x = dram_xs[dram];
-                for(i = 0; i < NUM_TESTS; ++i){
+                for(i = 0; i < NUM_TESTS; ++i) {
                         addr = addrs[i];
```
Comment (mrutt92) [inline regression/library/test_vcache_flush.cpp]: Suggestion; generate the random data here so that it is not reused between tests.

**Pull request merged**